src/content/en/updates/posts/2016/03/web-enabled-internet-of-things.markdown

```diff
@@ -0,0 +1,523 @@
+};
+{% endhighlight %}
+
+As I need to control the LED from the object, I add a
+`this._led member` which I use to store the Johnny-Five LED
+object. I also set the color of the LED to its' default
+value (white).
+
+{% highlight javascript %}
+board.on("ready", function() {
+  ...
+  colorCharacteristic._led = led;
+  led.color(colorCharacteristic._value);
+  led.intensity(30);
+  ...
+}
+{% endhighlight %}
+
+The "write" method received a string (just like "read" sends
+a string), which can consist of a CSS color code (ie. CSS names
+like `rebeccapurple` or hex codes like `#ff00bb`). I use a node
+module called parse-color to always get the hex value which is
+what Johnny-Five expects.
+
+{% highlight javascript %}
+ColorCharacteristic.prototype.onWriteRequest = function(data, offset, withoutResponse, callback) {
+  var value = parse(data.toString('utf8')).hex;
+  if (!value) {
+    callback(this.RESULT_SUCCESS);
+    return;
+  }
+    
+  this._value = value;
+  console.log(value);
+
+  if (this._led)
+      this._led.color(this._value);
+  callback(this.RESULT_SUCCESS);
+};
+{% endhighlight %}
+
+All the above will not work if we don't include the bleno module.
+Recall in the notes above that `eddystone-beacon` will not work
+with bleno unless you use the noble version distributed with it?
+Doing that is luckily quite simple:
+
+{% highlight javascript %}
+var bleno = require('eddystone-beacon/node_modules/bleno');
+var util = require('util');
+{% endhighlight %}
+
+Now all we need it to advertise our device (UUID) and its
+characteristics (other UUIDs)
+
+{% highlight javascript %}
+bleno.on('advertisingStart', function(error) {
+    ...
+    bleno.setServices([
+      new bleno.PrimaryService({
+        uuid: 'fc00',
+        characteristics: [
+          temperatureCharacteristic, colorCharacteristic
+        ]
+      })
+    ]);
+});
+{% endhighlight %}
+
+# Creating the client web app
+
+I don't want to go into too much details on how the non-bluetooth
+parts of the client app works, but I can tell you that I have used
+Polymer for easily creating a responsive user interface which serves
+the purpose of this demo. The resulting app looks like the below:
+
+<img alt="Client app on phone" src="/web/updates/images/2016/03/web-enabled-internet-of-things/phone-ux.png" style="width: 50%; float: left"/>
+<img alt="Error message" src="/web/updates/images/2016/03/web-enabled-internet-of-things/error.png" style="width: 50%; float: left"/>
+<br/>
+
+The right side shows an earlier version, show casing a simple error
+log that I added, easing the development.
+
+Web Bluetooth makes it dead easy to talk to a Bluetooth Low Energy
+device, so let's look at a simplified version of my connection code.
+If you don't know how promises work, please check out this resource
+before reading further:
+
+https://developers.google.com/web/fundamentals/primers/promises
+
+Connecting to a Bluetooth device involved a chain of promises.
```

Comment:
the url you advertise can have the MAC address in it I think at which point you don't have to search for the device... I think... @beaufortfrancois can probably tell me of I'm right or wrong here. I know that's possible with the FirefoxOS version of Web Bluetooth